File: Catheter-Control-main/src/maxon_epos_example/node/sample.cpp
/**
 * @file   sample
 * @brief  
 * @author 
 * @date    
 */

#include <ros/ros.h>
#include <std_msgs/Float32MultiArray.h>
#include <std_msgs/Float32MultiArray.h>
#include "maxon_epos_msgs/MotorState.h"

int main(int argc, char** argv)
{
    ros::init(argc, argv, "sample");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<std_msgs::Float32MultiArray>("/maxon_bringup/all_position", 1000);
    std_msgs::Float32MultiArray msg = std_msgs::Float32MultiArray();
    msg.data.push_back(3000);
    msg.data.push_back(-2000);

    // ros::Publisher pub_all_motors = nh.advertise<std_msgs::Float32MultiArray>("~/get_all_state", 1000);
    // maxon_epos_msgs::MotorStates msg = maxon_epos_msgs::MotorStates();
    
    ros::Rate sleep_rate(50);
    while (ros::ok()) {
        pub.publish(msg);
        sleep_rate.sleep();
    }
    
    return 0;
}

File: Catheter-Control-main/src/maxon_epos_example/node/sample.py
import rospy
from std_msgs.msg import Float32MultiArray
from maxon_epos_msgs.msg import MotorState


class CatheterController():
    def __init__(self):
        rospy.on_shutdown(self.shutdown)
        rospy.init_node("test_position_control", anonymous=True)

        self.initNode()
    
    def shutDown(self):
        rospy.loginfo("Stopping instant of EPOS...")
    

    def initNode(self):
        self.pub = rospy.Publisher("/maxon_bringup/all_position", MotorState, queue_size=1)
        


if __name__ == "__main__":

    cctrl = CatheterController()
    cctrl.pub.publish(MotorState(data=[10000, -20000]))
    
